keps/sig-cli/kustomize-exec-secret-generator.md

```diff
@@ -0,0 +1,190 @@
+  remote source without writing the Secrets as files to local disk.
+- Alice's organization configures the gitops deployment container to run kustomize in the cluster
+  and be capable of pulling Secrets from remote locations
+- Alice writes her kustomization.yaml to use the generation options configured by her organization.
+
+Example exploit to avoid:
+
+- Alice wants to run a whitebox mysql instance on a test cluster
+- Carl publishes a whitebox mysql `kustomization.yaml` on GitHub, with a SecretGenerator
+  that will read Alice's ~/.kube/config and send it to Carl's server by executing `sh`
+  will a script to generate some Secret
+- Alice runs `kubectl apply -k https://github.com/carl/mysql` and has the credentials
+  of all of her Kubernetes clusters sent to Carl when the Secret is generated.
+
+See [692][execRemoval] for more details.
+
+## Motivation
+
+The ability to create Kubernetes Secrets generated by commands is a commonly requested user.
+This is useful for use cases where the user doesn't want the Secrets to be appear decrypted
+on disk before being applied to the cluster.
+
+Examples:
+
+- Secrets sources are encrypted and need to be decrypted before they are applied
+- Secrets sources are not stored locally, and need to be fetched from some remote location
+- Secrets are generated using some function (e.g. private-keys)
+
+**Note:** For the common targeted case, *the solution for generating the Secret already
+exists as a command on the user's machine*.  User's are not expected to require a
+marketplace of solutions, rather instead they are expected to need to be able to
+invoke the tools they already depend upon.
+
+### Goals
+
+- Enable users to generate Secrets using the tools they already have on their machines
+- Require Alice to opt-in inorder for Carl to execute commands on her machine
+- Support Linux / Mac / Windows platforms
+
+### Non-Goals
+
+- Enable an ecosystem of users authoring plugins and sharing them
+- Reduce friction for publicly published whitebox `kustomization.yaml`s to generate Secrets
+
+
+## Proposal
+
+Re-introduce `exec` Secret generators, but require the executables to have a whitelisted prefix
+and require a flag.
+
+- Introduce a flag `--secret-generator-exec-prefix` defaulted to `kustomize-sg-`
+- Introduce a flag `--enable-exec-secret-generator` defaulted to `false`
+- Commands provided to generate Secrets *must* start with the prefix, or kustomize will exit 1
+- The enable flag *must* be provided if an exec Secret Generator is specified, or kustomize will exit 1
+- Command is executed from the `PATH` variable, it cannot be an absolute or relative path
+  - This is to prevent Carl from publishing a malicious generator with the kustomization.yaml and invoking it
+- Users can override the prefix with a string value with `len(value) > 2` - e.g. `sg-`
+  - Min length is to prevent users from providing an empty value, which would allow arbitrary command execution
```

Comment:
This seems a bit nanny-state imo.  If I have gone out of my way to overridde an option, then the tool should do-what-i-said (don't introduce an arms race with the user).

Reply:
@anguslees sorry to see that you aren't the author on this KEP.  Feel free to write a different one - there's no limit.
@pwittrock can we leave some of the more obvious KEP opportunities to new community members?

Reply:
I am fine with removing this.

Reply:
Yeah, maybe I'll start writing KEPs for KEPs :)

Reply:
@anguslees A colleague suggested my previous comment was inappropriate, and I apologize. I meant it literally.  You made some great points on #811, which strongly influenced its final form, and i hoped you or someone else who was interested in exec would contribute something along these lines.